optax/losses/_self_supervised_test.py

```diff
@@ -53,5 +53,69 @@ def test_batched(self):
     )
 
 
+class TripletMarginLossTest(chex.TestCase):
+
+  def setUp(self):
+    super().setUp()
+    self.a1 = jnp.ones((2, 2))
+    self.p1 = jnp.zeros((2, 2))
+    self.n1 = jnp.ones((2, 2))*2
+    self.a2 = jnp.zeros((2, 2))
+    self.p2 = jnp.ones((2, 2))
+    self.n2 = jnp.ones((2, 2))*2
+
+  @chex.all_variants
+  def test_batched(self):
+    def testing_triplet_loss(a, p, n, margin=1.0, swap=False):
+      ap = jnp.linalg.norm(a - p)
+      an = jnp.linalg.norm(a - n)
+      if swap:
+        pn = jnp.linalg.norm(p - n)
+        an = min(an, pn)
+      return jnp.maximum(ap - an + margin, 0)
+    handmade_result = testing_triplet_loss(
+    a=self.a1, p=self.p1, n=self.n1, margin=1.0, swap=False)
+    result = self.variant(_self_supervised.triplet_margin_loss)(
+    self.a1, self.p1, self.n1)
+    np.testing.assert_allclose(result, handmade_result, atol=1e-4)
+
+    handmade_result = testing_triplet_loss(
+    a=self.a2, p=self.p2, n=self.n2, margin=1.0, swap=False)
+    result = self.variant(_self_supervised.triplet_margin_loss)(
+    self.a2, self.p2, self.n2)
+    np.testing.assert_allclose(result, handmade_result, atol=1e-4)
+
+    handmade_result = testing_triplet_loss(
+    a=self.a1, p=self.p1, n=self.n1, margin=1.0, swap=True)
+    result = self.variant(_self_supervised.triplet_margin_loss)(
+    anchor=self.a1, positive=self.p1, negative=self.n1, swap=True)
+    np.testing.assert_allclose(result, handmade_result, atol=1e-4)
+
+    handmade_result = testing_triplet_loss(
+    a=self.a2, p=self.p2, n=self.n2, margin=1.0, swap=True)
+    result = self.variant(_self_supervised.triplet_margin_loss)(
+    anchor=self.a2, positive=self.p2, negative=self.n2, swap=True)
+    np.testing.assert_allclose(result, handmade_result, atol=1e-4)
+    original_loss = _self_supervised.triplet_margin_loss(
+                    self.a1, self.p1, self.n1)
+
+    # JIT compiled function result
+    jit_loss = (self.variant(jax.jit(
```

Comment:
No need to test jit if you already used self.variant.
self.variant is either the function as is or its jit version etc...